src/steps/video-setup/source-select.tsx

```diff
@@ -44,11 +44,20 @@ export const SourceSelection: React.FC<SourceSelectionProps> = ({
 
   const clickBoth = async () => {
     setActiveSource("both");
-    await startUserCapture(dispatch, settings, userConstraints);
-    await Promise.all([
-      queryMediaDevices(dispatch),
-      startDisplayCapture(dispatch, settings, displayConstraints),
-    ]);
+    // Safari needs the display capture to be called first as
+    // otherwise `getDisplayMedia` won't register the button click and
+    // throw an error.
+    if (navigator.userAgent.match(/safari/i) && !navigator.userAgent.match(/chrome/i)) {
```

Comment:
We do have the `onSafari` helper function in `util/index.tsx`. You can probably use that?

Reply:
Nice, that certainly is a useful helper^^